### ISSUE TITLE
Logged in but unauthorized

### ISSUE THREAD

**Issue by ziguane** [micronaut-projects/micronaut-core#882]
### Steps to Reproduce

1. Clone repository git clone https://github.com/ziguane/sample2.git
2. Change directory cd sample2 
3. Launch the app cd ./gradlew run
4. Visit the secured link http://localhost:8080 (you will redirect to /unauthorized)
5. Log in with username 'login' and password 'pwd' (you will be redirect to /)
6. click on the link to visit the secured link /periods/list/my (you are redirect to /unauthorized)

### Expected Behaviour

No redirection to /unauthorized when the user is logged in

### Actual Behaviour
The user is redirect as if he wasn't identified

### Environment Information

- **Operating System**: 10.12.6
- **Micronaut Version:** 1.0.0
- **JDK Version:** 1.8.0_102

### Example Application

- https://github.com/ziguane/sample2

FYI, already described in https://stackoverflow.com/q/53211440/10552571

**Comment by sdelamo** [comment]
@ziguane since `/peridos/list/my` does not match any route. would you consider the appropriate behaviour?

**Comment by sdelamo** [comment]
by the way which browser are you using. which is your OS? windows? MacOS?

**Comment by sdelamo** [comment]
You need to remove all the cookies for the domain in your browser.

E.g. in MacOS Mojava Safari Version 12.0.1 (14606.2.104.1.1) you need to do:

<img width="775" alt="screenshot_12_11_2018__10_33" src="https://user-images.githubusercontent.com/864788/48338614-675cfc80-e666-11e8-9a58-4af279c199e9.png">

Then, I will recommend to setup `session.http.cookiePath=/`. E.g.:

```
  session:
        http:
            cookiePath: /
```
Without that, in firefox I was getting two session cookies:

<img width="1006" alt="screenshot_12_11_2018__10_41" src="https://user-images.githubusercontent.com/864788/48339078-88721d00-e667-11e8-88c2-07ab5d26a18a.png">


Note, the previous behaviour was happing only if the NOT FOUND route was something with multiple levels. e.g. `/peridos/list/my`. If you used `/foo`, the issue with multiple cookies being created was not present. 

To sum up, removing the cookies from localhost and setting the `session.http.cookiePath=/` seems to redirect to forbidden page which I think it is the correct behaviour.

**Comment by ziguane** [comment]
@sdelamo 
There are many inaccuracies in my report.
I'am on macOs Sierra 10.12.6

'/periods/list/my' : I had not noticed that the controller had not been commits (fixed)

```
package sample2.controllers;

import java.security.Principal;
import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nullable;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;
import io.micronaut.views.View;

@Controller("/periods")
@Secured(SecurityRule.IS_ANONYMOUS)
public class PeriodLightController {
	
	@Get("/list/my")
	@View("periods/my")
	@Secured(SecurityRule.IS_AUTHENTICATED)
	Map<String, Object> myPeriods(@Nullable Principal principal) {
		Map<String, Object> data = new HashMap<>();
		return data;
	}
}

```

I set session.http.cookiePath=/ after a discussion with Mike Houston on Stackoverflow but nothing change until I remove all cookies for the domain. Now the behaviour seems to be ok. Mike had the same problems as me and has some improvement ideas for post login redirection http://michaelphouston.blogspot.com/2018/11/micronaut-and-redirect-on-security.html


Thanks for your help

**Comment by ziguane** [comment]
I thought the problem was solved, but it just changed. 
Previously, I set session.http.cookiePath=/ and remove cookies for the domain : things seems to be okay.
Now, after I logged in, visiting '/periods/list/myservice' and '/periods/list/ok' is OK but '/periods/list/my' -> redirect me to '/forbidden'....


```
package sample2.controllers;

import java.security.Principal;
import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nullable;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;
import io.micronaut.views.View;

@Controller("/periods")
@Secured(SecurityRule.IS_ANONYMOUS)
public class PeriodLightController {
	
	@Get("/list/my")
	@View("periods/my")
	@Secured(SecurityRule.IS_AUTHENTICATED)
	Map<String, Object> myPeriods(@Nullable Principal principal) {
		Map<String, Object> data = new HashMap<>();
		return data;
	}

	@Get("/list/myservice")
	@View("periods/service")
	@Secured(SecurityRule.IS_AUTHENTICATED)
	Map<String, Object> myServicePeriods(@Nullable Principal principal) {
		Map<String, Object> data = new HashMap<>();
		return data;
	}

	@Get("/list/all")
	@View("periods/all")
	@Secured(SecurityRule.IS_AUTHENTICATED)
	Map<String, Object> allPeriods(@Nullable Principal principal) {
		Map<String, Object> data = new HashMap<>();
		return data;
	}
}
```



Firefox 61.0.1 (64 bits)
macOs Sierra 10.12.6 (16G1618)
Micronaut Version: 1.0.0
JDK Version: 1.8.0_102

**Comment by jameskleeh** [comment]
@ziguane I can't reproduce the issue. After logging in I can visit all of the methods in the `PeriodController`.

**Comment by jameskleeh** [comment]
```
16:06:07.469 [pool-1-thread-5] DEBUG i.m.security.filters.SecurityFilter - Authorized request GET /periods/list/my. The rule provider io.micronaut.security.rules.SecuredAnnotationRule authorized the request.
16:06:40.988 [pool-1-thread-5] DEBUG i.m.security.filters.SecurityFilter - Authorized request GET /periods/list/myservice. The rule provider io.micronaut.security.rules.SecuredAnnotationRule authorized the request.
16:07:34.139 [pool-1-thread-5] DEBUG i.m.security.filters.SecurityFilter - Authorized request GET /periods/list/all. The rule provider io.micronaut.security.rules.SecuredAnnotationRule authorized the request.
```
